auto_py_to_exe/dialogs.py

```diff
@@ -40,10 +40,18 @@ def ask_file(file_type):
         file_path = askopenfilename(parent=root, filetypes=file_types)
     root.update()
 
-    if file_path == "":
-        return None
-    else:
-        return file_path
+    """
+    File dialog has 3 types of responses:
+     - An empty tuple '()', when no files/folders were selected;
+     - A single string '<filepath>', when a single file/folder was selected;
+     - A tuple of strings '(<filepath1, ..., filepathN>)', when a multiple files/folders were selected;
+
+    So bool() operator returns following values for the cases:
+     - bool(()) -> False;
+     - bool('<filepath>') -> True;
+     - bool(('<filepath1>', ..., '<filepathN>')) -> True;
+    """
```

Comment:
I've been trying to find the docs for the tkFileDialog/tkinter.filedialog functions but due to Python calling the tcl library, I haven't found the return types easy to find.

For me, just executing:

```python
from tkinter import Tk
from tkinter.filedialog import askopenfilename

root = Tk()
root.withdraw()
root.wm_attributes('-topmost', 1)
file_path = askopenfilename(parent=root)
root.update()
```

Will set `file_path` to an empty string if I do not select anything and a non-empty string if I do select a file. Yes, `bool("")` does return `False` but I am getting different results compared to this comment - when I cancel, I get an empty string (for `askopenfilenames` also).

What OS and version of Python are you using? This may explain the differences and may hint on what to look for. Also, did you get this comment from somewhere else? Maybe that could give some insight.

Reply:
Versions in my environment:
```
Host OS: Ubuntu 20.04 LTS
Python version: 3.8.5
tkinter.TkVersion: 8.6
```

I searched in the [docs](https://docs.python.org/3/library/dialog.html) and in the [sources](https://github.com/python/cpython/blob/3.9/Lib/tkinter/filedialog.py) and the only thing that comes to mind now is Tk version mismatch :grin:

Reply:
Might be OS differences - that is ok, I can update the comment later 👍